### PR TITLE
feat: lb product zoned

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: scaleway/data_source_lb_ip.go [hunk 1]
```diff
@@ -35,15 +35,15 @@ func dataSourceScalewayLbIP() *schema.Resource {
 }
```
Comment (jerome-quere) [inline scaleway/data_source_lb_ip.go:35]: Should datasource schema also be versionned and migrated ?

File: scaleway/resource_lb.go [hunk 2]
```diff
@@ -2,12 +2,15 @@ package scaleway
...
 import (
 	"context"
+	"fmt"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/scaleway/scaleway-sdk-go/api/lb/v1"
 	"github.com/scaleway/scaleway-sdk-go/scw"
+
```
Comment (jerome-quere) [inline scaleway/resource_lb.go:12]: extra line

File: scaleway/resource_lb.go [hunk 3]
```diff
@@ -22,7 +25,17 @@ func resourceScalewayLb() *schema.Resource {
...
-		SchemaVersion: 0,
+		SchemaVersion: 1,
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Version: 0,
+				Type: cty.Object(map[string]cty.Type{
+					"id": cty.String,
+					"ip_id": cty.String,
+				}),
+				Upgrade: resourceScalewayLbUpgradeIPID,
```
Comment (jerome-quere) [inline scaleway/resource_lb.go:36]: Should we call func resourceScalewayLbUpgradeV0 ?

File: scaleway/resource_lb.go [hunk 4]
```diff
@@ -185,3 +206,42 @@ func resourceScalewayLbDelete(ctx context.Context, d *schema.ResourceData, meta
 
 	return nil
 }
+
+//resourceScalewayLbUpgradeZoned allow upgrade the zoned resource. Note: please create another method for future upgrades not related
+// from version 0 to 1.
+func resourceScalewayLbUpgradeIPID(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	select {
```
Comment (jerome-quere) [inline scaleway/resource_lb.go:213]: Why do we need a select here ?

File: scaleway/resource_lb_backend.go [hunk 5]
```diff
@@ -237,20 +249,24 @@ func resourceScalewayLbBackend() *schema.Resource {
...
 func resourceScalewayLbBackendCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	lbAPI := lbAPI(meta)
+	lbAPI, defaultZone, err := lbAPIWithZone(d, meta)
 
-	region, LbID, err := parseRegionalID(d.Get("lb_id").(string))
+	zone, LbID, err := parseZonedID(d.Get("lb_id").(string))
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	// set default zone
```
Comment (jerome-quere) [inline scaleway/resource_lb_backend.go:258]: lb_id should always be a localised it. So we should never encounter this condition

**Review by jerome-quere — CHANGES_REQUESTED** [review]

File: scaleway/resource_lb.go (hunk 4 above)
Comment (Monitob) [reply scaleway/resource_lb.go:213]: When Done is used we the the select statements. https://golang.org/pkg/context/#Context

File: scaleway/data_source_lb_ip.go (hunk 1 above)
Comment (Monitob) [reply scaleway/data_source_lb_ip.go:39]: dataSource updated

**Review by jerome-quere — APPROVED** [review]

**Pull request merged**